src/MaterialDesignThemes.Wpf/DialogHost.cs

```diff
@@ -405,6 +409,50 @@ private static void IsOpenPropertyChangedCallback(DependencyObject dependencyObj
         })));
     }
 
+    
+    private void ListenForWindowStateChanged(Window? window)
+    {
+        window ??= Window.GetWindow(this);
+
+        if (window is not null)
+        {
+            window.StateChanged += Window_StateChanged;
+        }
+    }
+
+    private void Window_StateChanged(object? sender, EventArgs e)
+    {
+        if (sender is not Window window)
+        {
+            return;
+        }
+
+        var windowState = window.WindowState;
+        if (windowState == WindowState.Minimized)
+        {
+            _lastFocusedDialogElement = FocusManager.GetFocusedElement(window);
+            _previousWindowState = windowState;
+            return;
+        }
+
+        // We only need to focus anything manually if the window changes state from Minimized --> (Normal or Maximized)
+        // Going from Normal --> Maximized (and vice versa) is fine since the focus is already kept correctly
+        if (IsWindowRestoredFromMinimized() && IsLastFocusedDialogElementFocusable())
+        {
+            // Kinda hacky, but without a delay the focus doesn't always get set correctly because the Focus() method fires too early
+            Task.Delay(50).ContinueWith(_ => this.Dispatcher.BeginInvoke(DispatcherPriority.Background, new Action(() =>
+            {
+                _lastFocusedDialogElement!.Focus();
+            })));
+        }
```

Comment:
Because Focusable and Visible can change I would move this check inside of the Task.Delay.


```suggestion
        if (IsWindowRestoredFromMinimized())
        {
            // Kinda hacky, but without a delay the focus doesn't always get set correctly because the Focus() method fires too early
            Task.Delay(50).ContinueWith(_ => this.Dispatcher.BeginInvoke(DispatcherPriority.Background, new Action(() =>
            {
                if (IsLastFocusedDialogElementFocusable())
                {
                    _lastFocusedDialogElement!.Focus();
                }
            })));
        }
```